src/main/kotlin/com/cognifide/gradle/aem/common/instance/InstanceIMSClient.kt

```diff
@@ -0,0 +1,163 @@
+package com.cognifide.gradle.aem.common.instance
+
+import com.cognifide.gradle.aem.AemExtension
+import io.jsonwebtoken.Jwts
+import io.jsonwebtoken.SignatureAlgorithm
+import org.bouncycastle.openssl.PEMKeyPair
+import org.bouncycastle.openssl.PEMParser
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter
+import org.gradle.internal.impldep.com.google.api.client.http.HttpMethods
+import org.json.JSONObject
+import java.io.*
+import java.net.URL
+import java.nio.file.Files
+import java.security.KeyFactory
+import java.security.KeyPair
+import java.security.interfaces.RSAPrivateKey
+import java.security.spec.KeySpec
+import java.security.spec.PKCS8EncodedKeySpec
+import javax.net.ssl.HttpsURLConnection
+
+/**
+ * Generates Bearer token based on the credentials fetched from AEMaaCS
+ * by communicating with Adobe Identity Management Services.
+ */
+class InstanceIMSClient(private val aem: AemExtension) {
+
+    private val common = aem.common
+
+    /**
+     * URI pointing to key file copied from AEMaaCS console.
+     */
+    val keyPath = aem.obj.string {
+        aem.prop.string("instance.default.keyPath")?.let { set(it) }
+    }
+
+    /**
+     * Directory storing the fetched secret file from AEMaaCS instance.
+     */
+    private val secretDir = aem.obj.dir {
+        convention(aem.obj.buildDir("instance/secret"))
+    }
+
+    private val serviceTokenFile: File? get() = keyPath.orNull?.let {
+        common.fileTransfer.downloadTo(it, secretDir.get().asFile)
+    }
+
+    private lateinit var privateKey: String
```

Comment:
overused `lateinit`

Reply:
I didn't get that, what should I replace it with?